### PR TITLE
Update odo to use go 1.18

### LINKED ISSUE #6090: Upgrade odo to use go 1.18
/kind user-story

**NOTE**: this issue has been modified to update odo to use go 1.18. Refer https://github.com/redhat-developer/odo/issues/6090#issuecomment-1254819875 for additional context.

## User Story
As an odo-dev I want to odo to be work with Go 1.19 so that if I want to use the latest version, it does not break odo. There are a few discrepancies between the currently supported version and 1.19.

odo currently uses go 1.17, support for which ended with go1.19

## Acceptance Criteria 
- [ ] It should not break checks from the openshift-presubmit-unittest.
- [ ] release scripts end brew environment needs to be updated and verified that Go 1.19 can be used for building release artifacts 
- [ ] check what's supported by the Build team 

## Links 
- Related Epic (mandatory): 

/kind user-story

### PR DESCRIPTION
**Pull request by dharmit**
Signed-off-by: Dharmit Shah <shahdharmit@gmail.com>

**What type of PR is this:**
/kind feature

**What does this PR do / why we need it:**
$subject

**Which issue(s) this PR fixes:**

Fixes #6090

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
* I tested `make validate` and `make test` locally before opening the PR. Not sure if there was any other Makefile target that I should have tested locally.
* For a list of steps I followed, PTAL https://github.com/redhat-developer/odo/issues/6090#issuecomment-1255199161.

### CODE REVIEW

**Comment by anandrkskd** [comment]
@dharmit, just wanted to know if we have checked with EXD team if they have go 1.18 on rhel8, I remember we have go1.18 available for rhel7.

File: pkg/odo/util/cmdutils.go
Comment (feloy) [inline pkg/odo/util/cmdutils.go:10]: stdlib imports ("os" here) should appear in the first block of imports

File: tests/integration/cmd_namespace_test.go
Comment (feloy) [inline tests/integration/cmd_namespace_test.go:6]: Non stdlib imports should not appear in first block of imports

**Review by feloy — CHANGES_REQUESTED** [review]

**Comment by dharmit** [comment]
> @dharmit, just wanted to know if we have checked with EXD team if they have go 1.18 on rhel8, I remember we have go1.18 available for rhel7.

I have pinged Matej for this. I checked using the `brew` cli, but couldn't get the answer myself.

File: pkg/odo/util/cmdutils.go
Comment (dharmit) [reply pkg/odo/util/cmdutils.go]: IDE did something. I changed some settings, and hope it's not a problem any more. At least, I couldn't find any as far as I checked. Thanks for catching. 👍🏾

File: tests/integration/cmd_namespace_test.go
Comment (dharmit) [reply tests/integration/cmd_namespace_test.go]: https://github.com/redhat-developer/odo/pull/6166#discussion_r985388275

**Comment by anandrkskd** [comment]
> I have pinged Matej for this. I checked using the brew cli, but couldn't get the answer myself.

I checked for 1.18 on rhel8 but was not able to find it. for rhel 7 its available AFAIR

**Comment by dharmit** [comment]
> > I have pinged Matej for this. I checked using the brew cli, but couldn't get the answer myself.
> 
> I checked for 1.18 on rhel8 but was not able to find it. for rhel 7 its available AFAIR

Indeed. That's exactly what I found. But I saw the list had only up to go 1.10, which seems weird to me because it's too old. That's why I have pinged Matej about it since he would know the best. :)

**Comment by feloy** [comment]
@dharmit All tests fail, did you rebase after this PR? https://github.com/redhat-developer/odo/pull/6164

**Comment by dharmit** [comment]
> @dharmit All tests fail, did you rebase after this PR? #6164

I don't remember if I did or didn't. Anyway, rebased now.

**Comment by dharmit** [comment]
/hold

Putting this on hold to prevent accidental merge till Red Hat Release Engg team says OK to update to 1.18. In the meantime, I'll try to address CI issues and the folks can review/approve it so that it can be merged once we have a go ahead.

**Comment by dharmit** [comment]
> /hold
> 
> Putting this on hold to prevent accidental merge till Red Hat Release Engg team says OK to update to 1.18. In the meantime, I'll try to address CI issues and the folks can review/approve it so that it can be merged once we have a go ahead.

/hold cancel

Internal systems at RH have been updated to use go 1.18.

File: .ibm/images/Dockerfile
Comment (rm3l) [inline .ibm/images/Dockerfile:1]: @dharmit Just for double-checking (as I am not sure when this image is being built and pushed for our CI), I just tried to build this image, and the `go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045` command did not pass:

```
$ podman image build -t redhat-developer/odo-ibm-pipeline .ibm/images

...
STEP 4/5: USER tester
--> 1c5c6b2312b
STEP 5/5: RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service
go: go.mod file not found in current directory or any parent directory.
        'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
Error: error building at STEP "RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service": error while running runtime: exit status 1

```

**Review by rm3l — CHANGES_REQUESTED** [review]

File: .ibm/images/Dockerfile
Comment (rm3l) [reply .ibm/images/Dockerfile:1]: After a quick chat with Philippe, that is the image we use in the configuration of the stages for our IBM pipelines. And it is currently built and pushed manually to `quay.io/odo-dev/ibmcloudtest-image:latest`.
The container spawned by IBM Cloud to run our tests is then created off of this image.

BTW, the image could build successfully using `golang:1.17` as base image, but with a warning:
```
STEP 4/5: USER tester
--> e9a2883e72f
STEP 5/5: RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service
go: downloading github.com/kadel/odo-robot v0.0.0-20211217142858-965ea0dd8488
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
go: downloading github.com/google/go-github/v41 v41.0.0
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading google.golang.org/appengine v1.6.7
go: downloading golang.org/x/crypto v0.0.0-20210817164053-32db794688a5
go: downloading github.com/google/go-querystring v1.1.0
go: downloading github.com/golang/protobuf v1.4.2
go: downloading google.golang.org/protobuf v1.25.0
go get: installing executables with 'go get' in module mode is deprecated.
        Use 'go install pkg@version' instead.
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
Looking up 'cloud-object-storage' from repository 'IBM Cloud'...
Plug-in 'cloud-object-storage 1.6.0' found in repository 'IBM Cloud'
Attempting to download the binary file...
 16.06 MiB / 16.06 MiB  100.00% 1s01s
16842752 bytes downloaded
Installing binary...
OK
```

So I guess we need to install using `go install` instead..

Comment (feloy) [reply .ibm/images/Dockerfile:1]: This image is used for CI Pipelines on IBM Cloud.
Effectively, `go get` on this context does not work anymore with go 1.18, you will need to replace it with `go install`

Comment (dharmit) [reply .ibm/images/Dockerfile:1]: > So I guess we need to install `odo-robot` using `go install` instead..

> Effectively, go get on this context does not work anymore with go 1.18, you will need to replace it with go install

I should have read the error message closely. I only replied based on the part about "where the image is built" in @rm3l's initial comment. 

I have seen this error frequently since moving to go 1.18 locally and have pushed a fix.

Thanks @rm3l & @feloy

**Review by rm3l — APPROVED** [review]

**Comment by dharmit** [comment]
```
<< End Captured GinkgoWriter Output
  Unexpected error:
      <*url.Error | 0xc000170120>: {
          Op: "Get",
          URL: "http://127.0.0.1:45923/",
          Err: <*errors.errorString | 0xc0000b0100>{s: "EOF"},
      }
      Get "http://127.0.0.1:45923/": EOF
  occurred
  In [It] at: /go/src/github.com/redhat-developer/odo/tests/e2escenarios/e2e_test.go:32
------------------------------
```

overriding since tests are passing on IBM Cloud

/override ci/prow/v4.11-integration-e2e

**Pull request merged**
